fastapi-challenges/blog-api/devashmit/main.py

```
@@ -0,0 +1,67 @@
+    return post
+
+
+@app.post("/posts", response_model=PostOut, status_code=201)
+def create_post(body: PostCreate, db: Session = Depends(get_db)):
+    post = Post(**body.model_dump())
+    db.add(post); db.commit(); db.refresh(post)
+    return post
+
+
+@app.put("/posts/{post_id}", response_model=PostOut)
+def update_post(post_id: int, body: PostUpdate, db: Session = Depends(get_db)):
```

Ruff (0.15.15)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)